File: src/test/java/test/java/ExampleCmsManagerTest.java
package test.java;

import me.saro.dat.dat.DatCmsManager;
import me.saro.dat.dat.Payload;
import org.junit.jupiter.api.Test;

import java.io.IOException;

public class ExampleCmsManagerTest {

    @Test
    public void test() throws IOException, InterruptedException {
        // singleton
        DatCmsManager manager = DatCmsManager.builder()
                .uri("http://localhost:8088")
                //.intervalOff() // disable auto sync
                .intervalSeconds(1)
                .token("12345678901b")
                .build();

        // manual sync
        // manager.sync();

        try {
            String plain = "Unicode 유니코드 ユニコード 万国码 يونيكود यूनिकोड Юникод 🦄💻";
            String secure = "Ciphertext 암호문 暗号文 密文 Шифротекст Texte chiffré Geheimtext نص مشفر सिफरपाठ 🔐";

            System.out.println("plain : " + plain);
            System.out.println("secure : " + secure);

            // issue dat
            String dat = manager.issue(plain, secure).getOrThrow();
            System.out.println("dat : " + dat);

            // parse dat
            Payload payload = manager.parse(dat).getOrThrow();

            String payloadPlain = payload.getPlain();
            String payloadSecure = payload.getSecure();

            System.out.println("payload plain : " + payloadPlain);
            System.out.println("payload secure : " + payloadSecure);
        } catch (Exception e) {
            System.out.println("Ignore: is soft test: real connection test");
        }


        // wait
        Thread.sleep(5000);

    }
}

File: src/test/kotlin/test/kt/ExampleCmsManagerTest.kt
package test.kt

import me.saro.dat.dat.DatCmsManager.Companion.builder
import org.junit.jupiter.api.Test
import java.io.IOException

class ExampleCmsManagerTest {
    @Test
    @Throws(IOException::class, InterruptedException::class)
    fun test() {
        // singleton
        val manager = builder()
            .uri("http://localhost:8088")
            //.intervalOff() // disable auto sync
            .intervalSeconds(1)
            .token("12345678901b")
            .build()

        // manual sync
        // manager.sync();
        try {
            val plain = "Unicode 유니코드 ユニコード 万国码 يونيكود यूनिकोड Юникод 🦄💻"
            val secure = "Ciphertext 암호문 暗号文 密文 Шифротекст Texte chiffré Geheimtext نص مشفر सिफरपाठ 🔐"

            println("plain : " + plain)
            println("secure : " + secure)

            // issue dat
            val dat = manager.issue(plain, secure).getOrThrow()
            println("dat : " + dat)

            // parse dat
            val payload = manager.parse(dat).getOrThrow()

            val payloadPlain = payload.plain
            val payloadSecure = payload.secure

            println("payload plain : " + payloadPlain)
            println("payload secure : " + payloadSecure)
        } catch (e: Exception) {
            println("Ignore: is soft test: real connection test")
        }


        // wait
        Thread.sleep(5000)
    }
}
